privateSBOMExchange/tests/test_cpabe.py

```diff
@@ -0,0 +1,96 @@
+import configparser
+import copy
+import sys
+import os
+import configparser
+import filecmp
+from difflib import unified_diff
+from lib4sbom.parser import SBOMParser
+
+from ..src.petra.lib.models import *
+import cpabe
+"""This tests encryption and decryption based on policy and attributes 
+"""
+def show_differences(file1, file2):
+    with open(file1, 'r') as f1, open(file2, 'r') as f2:
+        f1_lines = f1.readlines()
+        f2_lines = f2.readlines()
+
+        diff = unified_diff(f1_lines, f2_lines, fromfile=file1, tofile=file2)
+        
+        for line in diff:
+            print(line, end='')
+
+def print_visitor_to_file(function, sbom_tree, filename):
+    """ Redirecting print output to a file"""
+    with open(filename, 'w') as file:
+        original_stdout = sys.stdout
+        try:
+            sys.stdout = file
+            
+            print("Printing raw SBOM tree")
+            print_visitor = function()
+            sbom_tree.accept(print_visitor)
+        finally:
+            sys.stdout = original_stdout
+
+def cleanup():
+    for file in ("build_output_tree.txt","hash_output_tree.txt","encrypt_output_tree.txt", "decrypt_output_tree.txt"):
+        if os.path.exists(file):
+            os.remove(file)
+
+
+config = configparser.ConfigParser()
+config.read('testConfigs/config.ini')
+target_sbom_dir = config['DEFAULT']['target_sbom']
+policy_files = [config['POLICY'][key] for key in ("intellectual_property_policy", "weaknesses_policy")]
+
+sample_short_sbom = "spdx-syft-ibmcom_kubedns-sha256:c12a28611a6883a2879b8f8dae6a7b088082d40262a116be51c8ee0b69cf91e0.json"
+sbom_file = os.path.join(target_sbom_dir, sample_short_sbom)
+
+Groups = ["Security Auditor", "Audit Authorization status of Approved"]
+pk, mk = cpabe.cpabe_setup();
+sk = cpabe.cpabe_keygen(pk, mk, Groups)
+
+# Parse SPDX data into a Document object
+SBOM_parser = SBOMParser()   
+SBOM_parser.parse_file(sbom_file) 
+SBOM_parser.parse_file("./sbom_data/bom-shelter-test/julia.spdx_short.json") 
+  
+sbom=SBOM_parser.sbom
+sbom_tree = build_sbom_tree(sbom)
+print_visitor_to_file(PrintFieldNode, sbom_tree,"build_output_tree.txt")
```

Comment:
I know we discussed that this is inefficient, but I am fine keeping this option around purely for debugging purposes.